### PR TITLE
MIGENG-185: Business Rules Implementation - WSR Complexity

### PR DESCRIPTION
**Pull request by carlosthe19916**
https://jira.coreos.com/browse/MIGENG-185
Added:
- `ComplexityModel`
- `ComplexityService`
- `ComplexityRepository`

Enhanced:
- `direct:calculate-workloadsummaryreportmodel` router
- `WorkloadSummaryReportRoutes_DirectCalculateVMWorkloadInventoryModelTest` test class

### CODE REVIEW

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/summary/ComplexityModel.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                        @ColumnResult(name = "medium", type = Integer.class),
+                        @ColumnResult(name = "difficult", type = Integer.class),
+                        @ColumnResult(name = "unknown", type = Integer.class)
+                }
+        )
+)
+
+@NamedNativeQuery(
+        name = "ComplexityModel.calculateComplexityModels",
+        query = "select sum(case when lower(complexity)='easy' then 1 else 0 end) as easy, sum(case when lower(complexity)='medium' then 1 else 0 end) as medium, sum(case when lower(complexity)='difficult' then 1 else 0 end) as difficult, sum(case when (complexity is null or lower(complexity)='unknown') then 1 else 0 end) as \"unknown\" from workload_inventory_report_model where analysis_id = :analysisId",
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/summary/ComplexityModel.java]:
```suggestion
        query = "select sum(case when lower(complexity)='easy' then 1 else 0 end) as easy, sum(case when lower(complexity)='medium' then 1 else 0 end) as medium, sum(case when lower(complexity)='hard' then 1 else 0 end) as hard, sum(case when (complexity is null or lower(complexity)='unknown') then 1 else 0 end) as \"unknown\" from workload_inventory_report_model where analysis_id = :analysisId",
```

**Review by mrizzi — COMMENTED** [review]
The overall code looks good (also the test) :+1: 
There's one change needed to be consistent with the required values for the complexity: instead of `difficult` we are required to use `hard`

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/summary/ComplexityModel.java (hunk 1 above)
Comment (carlosthe19916) [reply src/main/java/org/jboss/xavier/analytics/pojo/output/workload/summary/ComplexityModel.java]: Thanks for discovering this @mrizzi , I've already changed "difficult" by "hard" not just in the query but in the Class field.

File: src/main/java/org/jboss/xavier/analytics/pojo/output/workload/summary/ComplexityModel.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    public void setUnknown(Integer unknown) {
+        this.unknown = unknown;
+    }
+
+    @Override
+    public String toString() {
+        return "SummaryModel{" +
+                "id=" + id +
+                ", report=" + report +
+                ", easy='" + easy + '\'' +
```
Comment (mrizzi) [inline src/main/java/org/jboss/xavier/analytics/pojo/output/workload/summary/ComplexityModel.java]:
```suggestion
                ", easy='" + easy +
```

**Review by mrizzi — COMMENTED** [review]
Just a little change to clean the code.

**Pull request merged**